indico/modules/events/registration/models/registrations.py

```diff
@@ -126,6 +126,12 @@ class Registration(db.Model):
             lazy=True
         )
     )
+    #: If the registration has been deleted
+    is_deleted = db.Column(
```

Comment:
You need to adapt the two conditional unique indexes (`postgresql_where`) and add `NOT is_deleted` to their conditions.